File: src/api/dotnet/FPExpr.cs
/*++
Copyright (c) 2013 Microsoft Corporation

Module Name:

    FPExpr.cs

Abstract:

    Z3 Managed API: Floating Point Expressions

Author:

    Christoph Wintersteiger (cwinter) 2013-06-10

Notes:
    
--*/
using System.Diagnostics;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;


namespace Microsoft.Z3
{    
    /// <summary>
    /// FloatingPoint Expressions
    /// </summary>
    public class FPExpr : Expr
    {
        /// <summary>
        /// The number of exponent bits.
        /// </summary>
        public uint EBits { get { return ((FPSort)Sort).EBits; } }

        /// <summary>
        /// The number of significand bits.
        /// </summary>
        public uint SBits { get { return ((FPSort)Sort).SBits; } }

        #region Internal
        /// <summary> Constructor for FPExpr </summary>
        internal FPExpr(Context ctx, IntPtr obj)
            : base(ctx, obj)
        {
            Debug.Assert(ctx != null);
        }
        #endregion
    }
}

File: src/api/java/FPExpr.java
/*++
Copyright (c) 2013 Microsoft Corporation

Module Name:

    FPExpr.java

Abstract:

Author:

    Christoph Wintersteiger (cwinter) 2013-06-10

Notes:
    
--*/
package com.microsoft.z3;

/**
 * FloatingPoint Expressions
 */
public class FPExpr extends Expr<FPSort>
{   
    /**
     * The number of exponent bits.
     * @throws Z3Exception 
     */
    public int getEBits() { return ((FPSort)getSort()).getEBits(); }

    /**
     * The number of significand bits.
     * @throws Z3Exception 
     */
    public int getSBits() { return ((FPSort)getSort()).getSBits(); }

    /**
     * Indicates whether the floating-point expression is a numeral.
     * @throws Z3Exception on error
     * @return a boolean
     **/
    public boolean isNumeral()
    {
        return Native.fpaIsNumeral(getContext().nCtx(), getNativeObject());
    }

    public FPExpr(Context ctx, long obj)
    {
        super(ctx, obj); 
    }

}
